dev-utils/lint-tav-json.js

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+/*
+ * Copyright Elasticsearch B.V. and other contributors where applicable.
+ * Licensed under the BSD 2-Clause License; you may not use this file except in
+ * compliance with the BSD 2-Clause License.
+ */
+
+'use strict'
+
+// Check that the ".ci/tav.json" file includes all the modules defined in
+// the various ".tav.yml" files. If not, then CI "TAV" runs are not testing
+// all modules.
+//
+// Also check that the product of "versions" and "modules" in tav.json does
+// not exceed the 256 GH Actions limit.
+// https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits
+
+const fs = require('fs')
+const path = require('path')
+
+const glob = require('glob')
+const yaml = require('js-yaml')
+
+const TOP = path.resolve(path.join(__dirname, '..'))
+const TAV_JSON_PATH = path.join('.ci', 'tav.json')
+
+// ---- mainline
+
+function main (argv) {
+  let numErrors = 0
+
+  // Find all module names in ".tav.yml" files.
+  const moduleNamesFromYaml = new Set()
+  const tavYmlPaths = glob.sync(
+    '**/.tav.yml',
+    { ignore: ['**/node_modules/**'] }
+  )
+  // console.log('tavYmlPaths:', tavYmlPaths)
+  tavYmlPaths.forEach(p => {
+    const tavCfg = yaml.load(fs.readFileSync(p, 'utf8'))
+    Object.keys(tavCfg).forEach(k => {
+      const v = tavCfg[k]
+      moduleNamesFromYaml.add(v.name || k)
+    })
+  })
+  // console.log('moduleNamesFromYaml: ', moduleNamesFromYaml)
```

Comment:
nit: do you want to keep these logs?

Reply:
Kind of, yes. :)  If I ever have to come back and debug this script, then those would be the first interesting things I'd want to re-enable. However, granted, this is a simple script and I don't expect we'll need to come back and debug. I don't care that strongly either way.

To get waaay off track here: I sometime wonder at code (perhaps a language could have specific features/syntax for this) having ways of indicating points of interest for the debugging process.

Reply:
I guess the next (better?) step would be to add a logging lib, a command-line option and/or envvar to set the debug level, and make these log.debug statements.